src/System.Collections.NonGeneric/tests/Hashtable/AddTests.cs

```diff
@@ -0,0 +1,285 @@
+                {
+                    ht2.Add("KEY: aa==" + aa + " ,bb==" + bb, "VALUE: aa==" + aa + " ,bb==" + bb);
+                }
+
+                Assert.Equal(inLoops1, ht2.Count);
+                ht2.Clear();
+            }
+        }
+
+        const int iterations = 1600;
+
+        [Fact]
+        public void TestDuplicatedKeysWithInitialCapacity()
+        {
+            // Make rehash get called because to many items with dupplicated keys have been added to the hashtable
+            var ht = new Hashtable(200);
+
+            for (int i = 0; i < iterations; i += 2)
+            {
+                ht.Add(new BadHashCode(i), i.ToString());
+                ht.Add(new BadHashCode(i + 1), (i + 1).ToString());
+
+                ht.Remove(new BadHashCode(i));
+                ht.Remove(new BadHashCode(i + 1));
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                ht.Add(i.ToString(), i);
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                Assert.Equal(i, (int)ht[i.ToString()]);
+            }
+        }
+
+        [Fact]
+        public void TestDuplicatedKeysWithDefaultCapacity()
+        {
+            // Make rehash get called because to many items with duplicated keys have been added to the hashtable
+            var ht = new Hashtable();
+
+            for (int i = 0; i < iterations; i += 2)
+            {
+                ht.Add(new BadHashCode(i), i.ToString());
+                ht.Add(new BadHashCode(i + 1), (i + 1).ToString());
+
+                ht.Remove(new BadHashCode(i));
+                ht.Remove(new BadHashCode(i + 1));
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                ht.Add(i.ToString(), i);
+            }
+
+            for (int i = 0; i < iterations; i++)
+            {
+                Assert.Equal(i, (int)ht[i.ToString()]);
+            }
+        }
+    }
+
+    public class BadHashCode
+    {
+        private uint _value;
+
+        public BadHashCode(int value)
+        {
+            this._value = (uint)value;
+        }
+
+        public override bool Equals(object o)
+        {
+            BadHashCode rhValue = o as BadHashCode;
+
+            if (null != rhValue)
+            {
+                return _value.Equals(rhValue);
+            }
+            else
+            {
+                throw new ArgumentException("o", "is not BadHashCode type actual" + o.GetType());
+            }
+        }
+
+        public override int GetHashCode()
+        {
+            //The goal here is to return the same hashcode for different values
+            //First we clear the lowes order bits this will make every value even (0, 2, 4, 6, ...)
+            //However we would then be missing all of the odd values and we to return ever value
+            //So we divide by 2. (0, 2, 4, 6, ...) goes to (0, 1, 2, 3, ...)
+            //return (int)(value & (uint)0xFFFFFFFE) / 2;
+
```

Comment:
This comment block could probably just be removed, though possibly replaced with something like
// Return 0 for everything to force hash collisions.